retriever.py

```diff
@@ -61,102 +121,80 @@ def __init__(self, experiment_name):
-        # cosine similarity per chunk
-        # rank and print most/least match
-        tensor = torch.full(
-            (1, self.config.max_context_len),
-            self.tokenizer.pad_id,
-            dtype=torch.long,
-        )
-
-        functions = get_functions(
-            "/home/osalpekar/pytorch/torch/distributed/fsdp/fully_sharded_data_parallel.py",
-            indexing=False,
-            # "/home/osalpekar/pytorch/torch/fx/node.py"
-        )
-        for signature in functions:
-            # Just doing a sample function with the __init__ here
-            if "__init__" in signature:
-                function_body = functions[signature]
-                print(signature)
-                tokens = self.tokenizer.encode(function_body)  # .to("cuda:0")
-                tokens = tokens[: self.config.max_context_len]
-                tensor[0, : len(tokens)] = torch.tensor(
-                    tokens, dtype=torch.long
-                )
-                attn_mask = torch.where(tensor == self.tokenizer.pad_id, 0.0, 1.0)
-
-        tensor = tensor.to(self.device)
-        attn_mask = attn_mask.to(self.device)
+            grouped_by_file = defaultdict(list)
 
-        self.model.eval()
-        with torch.no_grad():
-            # full_model_states = self.model(
-            #     tokens, output_hidden_states=True
-            # )
-            # test_embedding = full_model_states.hidden_states[-1].detach()
-
-            _, func_embedding = self.model.forward(
-                tensor, 0, output_last_hidden_state=True, attn_mask=attn_mask
-            )
-            pooled_embedding = torch.sum(func_embedding, dim=1)
-
-            similarity_matrix = F.cosine_similarity(
-                self.embeddings, pooled_embedding
-            )
-            # Largest scores are last!
-            sorted_indices = torch.argsort(similarity_matrix, descending=False)
-
-            print("Top 10 Most Relevant Tests")
-            top_indices = sorted_indices[-10:]
-            for ind in top_indices:
-                test = self.unittest_names[int(ind.item())]
+            for ind in range(similarity_matrix.shape[0]):
+                test = self.unittest_names[ind]
                 score = similarity_matrix[ind]
-                print(f"Score: {score}; Test: {test}")
+                if test not in mapping:
+                    mapping[test] = []
+                mapping[test].append(score.item())
+
+        # condense
+        for test, score in mapping.items():
+            mapping[test] = sum(score) / len(score)
+        return mapping
 
-            print("Filtered Least Relevant Tests")
-            bot_indices = sorted_indices[:1000]
-            for ind in bot_indices:
-                test = self.unittest_names[int(ind.item())]
-                score = similarity_matrix[ind]
-                if "test" in test.split(":")[-1].lower():
-                    print(f"Score: {score}; Test: {test}")
+
+def gen_output_file(filename, mapping):
```

Comment:
This can be save_outputs func inside retriever. Let's keep the abstractions similar to indexer in terms of design choices.